tests/test_series.py

```diff
@@ -1726,48 +1746,114 @@ def test_updated_astype() -> None:
     check(assert_type(s.astype("Int16"), "pd.Series[int]"), pd.Series, np.int16)
     check(assert_type(s.astype("Int32"), "pd.Series[int]"), pd.Series, np.int32)
     check(assert_type(s.astype("Int64"), "pd.Series[int]"), pd.Series, np.int64)
+    # Pandas nullable unsigned integer types
+    check(assert_type(s.astype(pd.UInt8Dtype()), "pd.Series[int]"), pd.Series, np.uint8)
+    check(
+        assert_type(s.astype(pd.UInt16Dtype()), "pd.Series[int]"), pd.Series, np.uint16
+    )
+    check(
+        assert_type(s.astype(pd.UInt32Dtype()), "pd.Series[int]"), pd.Series, np.uint32
+    )
+    check(
+        assert_type(s.astype(pd.UInt64Dtype()), "pd.Series[int]"), pd.Series, np.uint64
+    )
+    check(assert_type(s.astype("UInt8"), "pd.Series[int]"), pd.Series, np.uint8)
+    check(assert_type(s.astype("UInt16"), "pd.Series[int]"), pd.Series, np.uint16)
+    check(assert_type(s.astype("UInt32"), "pd.Series[int]"), pd.Series, np.uint32)
+    check(assert_type(s.astype("UInt64"), "pd.Series[int]"), pd.Series, np.uint64)
+
     # Numpy signed integer types
+    # int8
     check(assert_type(s.astype(np.byte), "pd.Series[int]"), pd.Series, np.byte)
-    check(assert_type(s.astype(np.int8), "pd.Series[int]"), pd.Series, np.int8)
-    check(assert_type(s.astype(np.int16), "pd.Series[int]"), pd.Series, np.int16)
-    check(assert_type(s.astype(np.int32), "pd.Series[int]"), pd.Series, np.int32)
-    check(assert_type(s.astype(np.int64), "pd.Series[int]"), pd.Series, np.int64)
-    check(assert_type(s.astype(np.intp), "pd.Series[int]"), pd.Series, np.intp)
     check(assert_type(s.astype("byte"), "pd.Series[int]"), pd.Series, np.byte)
-    check(assert_type(s.astype("int8"), "pd.Series[int]"), pd.Series, np.int8)
-    check(assert_type(s.astype("int16"), "pd.Series[int]"), pd.Series, np.int16)
-    check(assert_type(s.astype("int32"), "pd.Series[int]"), pd.Series, np.int32)
-    check(assert_type(s.astype("int64"), "pd.Series[int]"), pd.Series, np.int64)
-    check(assert_type(s.astype("intp"), "pd.Series[int]"), pd.Series, np.intp)
+    check(assert_type(s.astype("int8"), "pd.Series[int]"), pd.Series, np.byte)
+    check(assert_type(s.astype("b"), "pd.Series[int]"), pd.Series, np.byte)
+    # int16
+    check(assert_type(s.astype(np.short), "pd.Series[int]"), pd.Series, np.short)
+    check(assert_type(s.astype("short"), "pd.Series[int]"), pd.Series, np.short)
+    check(assert_type(s.astype("int16"), "pd.Series[int]"), pd.Series, np.short)
+    check(assert_type(s.astype("h"), "pd.Series[int]"), pd.Series, np.short)
+    # int32
+    check(assert_type(s.astype(np.intc), "pd.Series[int]"), pd.Series, np.intc)
+    check(assert_type(s.astype("intc"), "pd.Series[int]"), pd.Series, np.intc)
+    check(assert_type(s.astype("int32"), "pd.Series[int]"), pd.Series, np.intc)
```

Comment:
Hm, I'm confused now: Reading https://numpy.org/doc/stable/reference/arrays.scalars.html it sounds like `intc` should be the platform independent type. https://numpy.org/doc/stable/reference/arrays.scalars.html#sized-aliases says

> Along with their (mostly) C-derived names, the integer, float, and complex data-types are also available using a bit-width convention so that an array of the right size can always be ensured.

Reply:
```python
>>> import pandas as pd
>>> import numpy as np
>>> s = pd.Series([1,2,3])
>>> s.dtype
dtype('int64')
>>> s32 = s.astype("int32")
>>> s32.dtype
dtype('int32')
>>> type(s32[0])
<class 'numpy.int32'>
>>> isinstance(s32[0], np.intc)
False
```
The above is on Windows.  When you use `s.astype("int32")`, then the resulting `int` is not an instance of `np.intc`.  On Linux and Mac, it is.  Not sure I understand why there is a difference, but it is what it is!

Reply:
I refactored the tests to use `pytest.mark.parametrize`, because I expect more windows tests to fail, and need to find out which. Also added "<M8[...]"  and "<m8[...]" type-codes for timestamp/timedelta.

Reply:
What happens on windows if you try `astype(np.intc)` ?

Reply:
> What happens on windows if you try `astype(np.intc)` ?
```python
>>> sc = s.astype(np.intc)
>>> sc.dtype
dtype('int32')
>>> type(sc[0])
<class 'numpy.intc'>
```

So I guess on Windows `s.astype("int32")` and `s.astype(np.intc)` store objects of different types inside the series.